addons/web_editor/static/src/js/backend/html_field.js

```diff
@@ -369,20 +369,37 @@ export class HtmlField extends Component {
     }
     async commitChanges({ urgent } = {}) {
         if (this._isDirty() || urgent) {
+            let toInlinePromise,
+                inlineSync = false,
+                toInlinePromiseIsResolved = false;
+            if (this.props.isInlineStyle && this.wysiwyg) {
+                // Avoid listening to changes made during the _toInline process.
+                this.wysiwyg.odooEditor.observerUnactive('commitChanges');
+                toInlinePromise =  this._toInline().then((value) => {
+                    toInlinePromiseIsResolved = true;
+                });
+            }
             if (urgent) {
+                // if the th inline promise is already complete,
+                // set inlineSync to true to avoid saving twice.
+                if (toInlinePromise && toInlinePromiseIsResolved) {
+                    inlineSync = true;
```

Comment:
I suppose I might be missing something but I don't understand how this condition can ever be true since `toInlinePromiseIsResolved` is set to `true` in a `.then` and is therefore always asynchronous, unless there is some magic `.then` function that _is_ synchronous ? I feel like this might be an artefact of the previous version of this PR where there were two different `toInline` functions.

Reply:
Indeed, good catch.

I wanted to avoid calling `updateValue` twice when the toInline process was synchomous. But i forgot to take the `then` into account. And the `then` will always force the async...

maybe there's another way to avoid updating the value twice, maybe it's not a big deal...

What do you think ?

Reply:
Mmmh... If I understand the problem correctly, I'm not sure it's going to be easy, if it even is possible at all. Therefore, it had me wondering. Is there really an issue to call `updateValue` twice ? Because if it isn't we might as well just let it.

Reply:
It's not an issue, but it might double the save call to the DB. 
Not even entirely sure, maybe OWL is clever about it.

Reply:
seems that owl only do one write call to the server. so I guess we are good.